### PR TITLE
Issue #3250215 by chmez: Make a form for adding a new member to the group re-usable

### PR DESCRIPTION
**Pull request by ghost**
## Problem
Altering of form for adding a new member isn't re-usable and a validator for checking an altered field works only with groups.

## Solution
Move code that alters form to service and add support to all entity types in the validator.

## Issue tracker
- https://www.drupal.org/project/social/issues/3250215
- https://getopensocial.atlassian.net/browse/YANG-6473

## How to test
- [x] When you add a field to your form through the "addMemberFormField" method of the "social_group.helper_service" service then you should see this field as a widget for selecting multiple users.

## Release notes
Make a form for adding a new member to the group re-usable.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2610-jsem5mqgnk7je5w5ug0fpihhmsazikx8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61976efce368654cf1421c5f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2610-jsem5mqgnk7je5w5ug0fpihhmsazikx8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61976efce368654cf1421c5f

**Review by ribel — APPROVED** [review]
Changes here looks good to me. I have checked that it doesn't cause a regression for the old group types.
@OH17L I think you can move it from Draft status?

**Comment by OH17L** [comment]
@ribel, done.

**Review by Ressinel — CHANGES_REQUESTED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2610-jsem5mqgnk7je5w5ug0fpihhmsazikx8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61976efce368654cf1421c5f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2610-jsem5mqgnk7je5w5ug0fpihhmsazikx8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61976efce368654cf1421c5f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2610-jsem5mqgnk7je5w5ug0fpihhmsazikx8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61976efce368654cf1421c5f

File: modules/social_features/social_group/src/SocialGroupHelperService.php [hunk 1]
```diff
@@ -254,54 +256,66 @@ public function countGroupMembershipsForUser($uid): int {
...
       // When the user has permission to create a group of the current type, add
-      // this to the create group array.
-      if ($account->hasPermission('create ' . $group_type->id() . ' group')) {
-        $user_can_create_groups[$group_type->id()] = $group_type;
-      }
-
-      if (count($user_can_create_groups) > 1) {
+      // this to the creation group array.
+      if ($account->hasPermission('create ' . $group_type . ' group')) {
+        $found = TRUE;
```
[the remark this answers is not in the document]
Comment (OH17L) [reply modules/social_features/social_group/src/SocialGroupHelperService.php]: Fixed.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2610-jsem5mqgnk7je5w5ug0fpihhmsazikx8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61976efce368654cf1421c5f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2610-jsem5mqgnk7je5w5ug0fpihhmsazikx8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61976efce368654cf1421c5f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2610-jsem5mqgnk7je5w5ug0fpihhmsazikx8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61976efce368654cf1421c5f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2610-jsem5mqgnk7je5w5ug0fpihhmsazikx8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61976efce368654cf1421c5f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2610-jsem5mqgnk7je5w5ug0fpihhmsazikx8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61976efce368654cf1421c5f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2610-jsem5mqgnk7je5w5ug0fpihhmsazikx8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61976efce368654cf1421c5f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2610-jsem5mqgnk7je5w5ug0fpihhmsazikx8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61976efce368654cf1421c5f

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2610-jsem5mqgnk7je5w5ug0fpihhmsazikx8.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61976efce368654cf1421c5f

**Comment by navneet0693** [comment]
@OH17L Please rebase the PR branch.

**Pull request merged**